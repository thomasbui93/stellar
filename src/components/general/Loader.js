import React from 'react';

export const Loader = () => {
  return <div className="app-loader">Loader</div>
};