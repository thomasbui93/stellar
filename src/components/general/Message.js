import React from 'react'
import PropTypes from 'prop-types'

export default class Message extends React.Component {
  static propTypes = {
    message: PropTypes.shape({
      type: PropTypes.oneOf(['error', 'notice']),
      info: PropTypes.string
    }).isRequired
  }

  state = {
    isClosed: false
  }

  render() {
    return (
      <div className={`${this.props.message.type} message ${this.state.isClosed ? 'hidden': ''}`}>
        <span>{this.props.message.info}</span>
        <span className="close-button" onClick={() => { this.setState({ isClosed: true })}}>
          <i className="material-icons">close</i>
        </span>
      </div>
    )
  }
}
