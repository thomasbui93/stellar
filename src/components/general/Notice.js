import React from 'react'

export default (props, context) => {
  return (
    <div className={`${props.type} notice-message`}>{props.message}</div>
  )
}
