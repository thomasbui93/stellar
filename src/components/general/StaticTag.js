import React from 'react';

export const StaticTag = (props) => {
  return <div className={`${props.className} static-tag`}>{props.staticText}</div>
};