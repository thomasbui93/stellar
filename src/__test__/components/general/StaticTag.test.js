import React from 'react'
import {StaticTag} from '../../../components/general/StaticTag'
import renderer from 'react-test-renderer'
import {shallow} from 'enzyme'

describe('StaticTag Component', function () {
  it('render correctly', () => {
    const component = renderer.create(<StaticTag />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render props correctly', () => {
    const wrapper = shallow(<StaticTag className='test-class' staticText='A static text' />)
    expect(wrapper.props().className).toEqual('test-class static-tag')
  })

  it('should render passed props into the component', () => {
    const wrapper = shallow(<StaticTag className='test-class' staticText='A static text' />)
    expect(wrapper.text()).toEqual('A static text')
  })
})
