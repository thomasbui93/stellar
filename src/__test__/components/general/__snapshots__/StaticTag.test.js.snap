// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StaticTag Component render correctly 1`] = `
<div
  className="undefined static-tag"
/>
`;
